cpp/src/lance/io/project.cc

```diff
@@ -80,10 +80,16 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> Project::Execute(
           std::static_pointer_cast<decltype(indices)::element_type>(indices->Slice(offset, len));
       values = values->Slice(offset, len);
     }
-    ARROW_ASSIGN_OR_RAISE(auto batch, reader->ReadBatch(*scan_schema_, batch_id, indices));
-    assert(values->num_rows() == batch->num_rows());
-    ARROW_ASSIGN_OR_RAISE(auto merged, lance::arrow::MergeRecordBatches(values, batch));
-    return merged;
+    if (scan_schema_->fields().empty()) {
```

Comment:
What happens if user passes in lance.scanner(ds, columns=[])? Will they get an empty table back or will an exception be raised?

Reply:
we have not had the check, but I feel that it should not be a silent error, so it should raise an exception.